### PR TITLE
Make multiprocessing Pool safe in quickquasars

### PR DESCRIPTION
**Pull request by andreicuceu**
The best practice for using multiprocessing.Pool is within a "with as" statement to safely terminate the pool.

We (with Naim) found a bug when nproc > nfiles. This is the error message at the end of running quickquasars:

Exception ignored in: <function Pool.__del__ at 0x1555439bcf70>
Traceback (most recent call last):
  File "/global/homes/a/acuceu/.conda/envs/desienv/lib/python3.9/multiprocessing/pool.py", line 268, in __del__
    self._change_notifier.put(None)
  File "/global/homes/a/acuceu/.conda/envs/desienv/lib/python3.9/multiprocessing/queues.py", line 378, in put
    self._writer.send_bytes(obj)
  File "/global/homes/a/acuceu/.conda/envs/desienv/lib/python3.9/multiprocessing/connection.py", line 205, in send_bytes
    self._send_bytes(m[offset:offset + size])
  File "/global/homes/a/acuceu/.conda/envs/desienv/lib/python3.9/multiprocessing/connection.py", line 416, in _send_bytes
    self._send(header + buf)
  File "/global/homes/a/acuceu/.conda/envs/desienv/lib/python3.9/multiprocessing/connection.py", line 373, in _send
    n = write(self._handle, buf)
OSError: [Errno 9] Bad file descriptor

This goes away when using "with multiprocessing.Pool(args.nproc) as pool:" to clean the pool after it's used.

### CODE REVIEW

**Review by sbailey — APPROVED** [review]
Very good!  We had previously had similar mystery failures that seemed to improve by using pool.close() after pool.map(), but using the `with multiprocessing.Pool(args.nproc) as pool:` context manager is even better.

**Comment by andreicuceu** [comment]
> @andreicuceu - why is this not merged? @alxogm ?

Not sure. The coverage test failed for some reason, and the output isn't available anymore. Can you rerun the automatic tests?

**Comment by alxogm** [comment]
> @andreicuceu - why is this not merged? @alxogm ?

It seems I missed this PR. I'm happy to merge if was already tested.

**Comment by alxogm** [comment]
@andreicuceu it seems the automatic tests can not be re-run at this stage, but you can make small push to trigger them again. Alternatively we can include your change in Hiram's new PR.

**Comment by andreicuceu** [comment]
@alxogm I added a small comment, so this should trigger another run of the automatic tests, but it's awaiting approval.

**Comment by andreicuceu** [comment]
@sbailey @alxogm It looks to me like the problem is because I did the pull request back when the main branch was still called "master" whereas now it's called "main". Not sure what to do. I could delete this pull request and start a new one on the most current version with the same change. It's just two lines of code after all.

**Comment by sbailey** [comment]
Yes, let's do a new PR off of current main.  Thanks.

**Pull request closed without merging**
